lib/elixir/lib/enum.ex

```diff
@@ -513,15 +525,15 @@ defmodule Enum do
     {_, {res, acc}} =
       Enumerable.reduce(enumerable, {:cont, {[], acc}}, fn entry, {buffer, acc} ->
         case chunk_fun.(entry, acc) do
-          {:cont, emit, acc} -> {:cont, {[emit | buffer], acc}}
+          {:cont, chunk, acc} -> {:cont, {[chunk | buffer], acc}}
           {:cont, acc} -> {:cont, {buffer, acc}}
           {:halt, acc} -> {:halt, {buffer, acc}}
         end
       end)
 
     case after_fun.(acc) do
       {:cont, _acc} -> :lists.reverse(res)
-      {:cont, elem, _acc} -> :lists.reverse([elem | res])
+      {:cont, chunk, _acc} -> :lists.reverse([chunk | res])
```

Comment:
This was the part that initially confused me, and took a while to grok what is going on here. The confusing part is that `_acc` is required in the returned tuple from `after_fun`, but it's ignored.

I'm wondering if it makes sense to deprecate that, and change the return shape to look something more like:

```elixir
case after_fun.(acc) do
  {:chunk, chunk} ->
    :lists.reverse([chunk | res])
  :ok ->
    :lists.reverse(res)
  {:cont, chunk, _acc} ->
    IO.warn("Enum.chunk_while/4 after_fun callback function returning {:cont, chunk, acc} is deprecated, please return {:chunk, chunk} instead")
    :lists.reverse(res)
  {:cont, _acc} ->
    IO.warn("Enum.chunk_while/4 after_fun callback function returning {:cont, acc} is deprecated, please return :ok instead")
    :lists.reverse(res)
end
```

Maybe also changing the return shape from `chunk_fun` to something similar too.